### DISCUSSION TITLE
error on making helloworld webpage on django

### DISCUSSION THREAD

**Discussion by XFrevert** [community/community#21370] [Other Feature Feedback, Questions, & Ideas]
<p>i kinda started it today and was creating a helloworld bebpage by reading in a book </p>
<p>so after doing all kind of work i mean you all know what kind of work i am talking about then </p>
<p>when i typed the command in power shell “python manage.py runserver”</p>
<p>it shows like this</p>
<hr>
<p>Unhandled exception in thread started by &lt;function check_errors.&lt;locals&gt;.wrapper at 0x0000027210AC3A60&gt;<br>
Traceback (most recent call last):<br>
File “C:\Users\A.virtualenvs\helloworld-PwcuqGam\lib\site-packages\django\utils\autoreload.py”, line 225, in wrapper<br>
fn(*args, **kwargs)<br>
File “C:\Users\A.virtualenvs\helloworld-PwcuqGam\lib\site-packages\django\core\management\commands\runserver.py”, line 109, in inner_run<br>
autoreload.raise_last_exception()<br>
File “C:\Users\A.virtualenvs\helloworld-PwcuqGam\lib\site-packages\django\utils\autoreload.py”, line 248, in raise_last_exception<br>
raise _exception[1]<br>
File “C:\Users\A.virtualenvs\helloworld-PwcuqGam\lib\site-packages\django\core\management__init__.py”, line 337, in execute<br>
autoreload.check_errors(django.setup)()<br>
File “C:\Users\A.virtualenvs\helloworld-PwcuqGam\lib\site-packages\django\utils\autoreload.py”, line 225, in wrapper<br>
fn(*args, **kwargs)<br>
File “C:\Users\A.virtualenvs\helloworld-PwcuqGam\lib\site-packages\django__init__.py”, line 24, in setup<br>
apps.populate(settings.INSTALLED_APPS)<br>
File “C:\Users\A.virtualenvs\helloworld-PwcuqGam\lib\site-packages\django\apps\registry.py”, line 89, in populate<br>
app_config = AppConfig.create(entry)<br>
File “C:\Users\A.virtualenvs\helloworld-PwcuqGam\lib\site-packages\django\apps\config.py”, line 123, in create<br>
import_module(entry)<br>
File “C:\Users\A.virtualenvs\helloworld-PwcuqGam\lib\importlib__init__.py”, line 126, in import_module<br>
return _bootstrap._gcd_import(name[level:], package, level)<br>
File “&lt;frozen importlib._bootstrap&gt;”, line 994, in _gcd_import<br>
File “&lt;frozen importlib._bootstrap&gt;”, line 971, in _find_and_load<br>
File “&lt;frozen importlib._bootstrap&gt;”, line 950, in _find_and_load_unlocked<br>
ModuleNotFoundError: No module named ‘pages.apps.pagesconfig’; ‘pages.apps’ is not a package</p>
<hr>
<p>i dont understand what kind of error is this plz help me about this thx in advance</p>

**Comment by fire-eggs** [comment, marked as answer]
<p>Django is looking for a set of files for an app called ‘pages’. That is, you presumably added the line</p>
<pre><code>'pages.apps.pagesconfig',
</code></pre>
<p>to your settings.py file. But Django cannot ‘find’ it, and thus the error.</p>
<p>There are a bunch of different reasons why Django might not be able to “find” it.</p>
<ol>
<li>
<p>Did you check that the case is correct? Any sample I see where the app “pages” is being created, the line to put in settings.py is </p>
<p>‘pages.apps.PagesConfig’,</p>
</li>
</ol>
<p>Note that PagesConfig has different case from what you probably entered.</p>
<ol start="2">
<li>
<p>Did you create an app called “pages”? I.e. did you do:</p>
<p>$ django-admin startproject pages_project .<br>
$ python manage.py startapp pages</p>
</li>
</ol>
<p>If you called your app something else, you have to put the appropriate line in settings.py.</p>
<p>You might try looking at <a href="https://djangoforbeginners.com/pages-app/" rel="nofollow noopener">this</a> or <a href="https://mywebhome.eu/2018/06/09/my-django-cheats/" rel="nofollow noopener">this </a>for some more details on the pages app. <a href="https://stackoverflow.com/questions/41963141/django-says-no-module-named-blog" rel="nofollow noopener">This</a> should give you some more background on your error.</p>
